pkg/ingestor/parser/open_vex/parser_open_vex.go

```diff
@@ -0,0 +1,152 @@
+		"fixed":               generated.VexStatusFixed,
+		"under_investigation": generated.VexStatusUnderInvestigation,
+	}
+)
+
+type openVEXParser struct {
+	identifierStrings *common.IdentifierStrings
+	openVex           *vex.VEX
+}
+
+func NewOpenVEXParser() common.DocumentParser {
+	return &openVEXParser{
+		identifierStrings: &common.IdentifierStrings{},
+	}
+}
+
+// Parse breaks out the document into the graph components
+func (c *openVEXParser) Parse(ctx context.Context, doc *processor.Document) error {
+	return json.Unmarshal(doc.Blob, &c.openVex)
+}
+
+// GetIdentities gets the identity node from the document if they exist
+func (c *openVEXParser) GetIdentities(ctx context.Context) []common.TrustInformation {
+	return nil
+}
+
+func (c *openVEXParser) GetIdentifiers(ctx context.Context) (*common.IdentifierStrings, error) {
+	return c.identifierStrings, nil
+}
+
+func (c *openVEXParser) generateVexIngest(vulnInput *generated.VulnerabilityInputSpec, vexStatement *vex.Statement, status string) ([]assembler.VexIngest, error) {
+	var vi []assembler.VexIngest
+
+	for _, p := range vexStatement.Products {
+		vd := generated.VexStatementInputSpec{}
+		vd.KnownSince = *c.openVex.Metadata.Timestamp
+		vd.Origin = c.openVex.Metadata.ID
+
+		ingest := assembler.VexIngest{}
+
+		if vexStatus, ok := vexStatusMap[status]; ok {
+			vd.Status = vexStatus
+		}
+
+		if vd.Status == generated.VexStatusNotAffected {
+			vd.Statement = vexStatement.ImpactStatement
+		} else if vd.Status == generated.VexStatusAffected {
+			vd.Statement = vexStatement.ActionStatement
+		}
+
+		vd.VexJustification = justificationsMap[string(vexStatement.Justification)]
+
+		ingest.VexData = &vd
+		ingest.Vulnerability = vulnInput
+
+		var err error
+		if ingest.Pkg, err = helpers.PurlToPkg(p); err != nil {
+			return nil, err
+		}
+
+		c.identifierStrings.PurlStrings = append(c.identifierStrings.PurlStrings, p)
+
+		vi = append(vi, ingest)
+	}
+
+	return vi, nil
+}
+
+func (c *openVEXParser) GetPredicates(ctx context.Context) *assembler.IngestPredicates {
+	logger := logging.FromContext(ctx)
+
+	var vis []assembler.VexIngest
+	var cvs []assembler.CertifyVulnIngest
+
+	for _, s := range c.openVex.Statements {
+		vuln, err := helpers.CreateVulnInput(s.Vulnerability)
+		if err != nil {
+			logger.Errorf("failed to create vulnerability input: %v", err)
+			continue
+		}
+
+		vi, err := c.generateVexIngest(vuln, &s, string(s.Status))
+		if err != nil {
+			logger.Errorf("failed to generate vex ingest: %v", err)
+			continue
+		}
+
+		for _, ingest := range vi {
+			vis = append(vis, ingest)
+
+			if s.Status == vex.StatusAffected || s.Status == vex.StatusUnderInvestigation {
+				vulnData := generated.ScanMetadataInput{
+					TimeScanned: *c.openVex.Metadata.Timestamp,
+				}
+				cv := assembler.CertifyVulnIngest{
+					Pkg:           ingest.Pkg,
+					Vulnerability: vuln,
+					VulnData:      &vulnData,
+				}
+				cvs = append(cvs, cv)
```

Comment:
change to
```go
c.cvs = append(c.cvs, cv)
```

Reply:
I have got it to start appending to `c` as well.